clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4119,20 +4119,26 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     }
   }
 
+  Arg *SYCLStdArg = Args.getLastArg(options::OPT_sycl_std_EQ);
+
```

Comment:
Maybe there's a way to move this `Arg` extraction before the preceding condition block (starting at line 4085), and then handle the `wno-sycl-strict` passing inside the mentioned block? This should reduce condition checks.

Reply:
@s-kanaev, any decision here?

Reply:
Done